### PR TITLE
pencil : fix udev dependency

### PR DESCRIPTION
**Pull request by mrVanDalo**
###### Motivation for this change

because of changes in the udev package the package was not working anymore

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/graphics/pencil/default.nix [hunk 1]
```diff
@@ -89,7 +89,7 @@ stdenv.mkDerivation rec {
     chmod a+x $out/opt/Pencil/libffmpeg.so
 
     # fix missing libudev
-    ln -s ${udev}/lib/systemd/libsystemd-shared.so $out/opt/Pencil/libudev.so.1
+    ln -s ${eudev}/lib/libudev.so.1 $out/opt/Pencil/libudev.so.1
```
Comment (Mic92) [inline pkgs/applications/graphics/pencil/default.nix]: Would `ln -s ${systemd.lib}/lib/libudev.so.1  $out/opt/Pencil/libudev.so.1` also work?

Comment (mrVanDalo) [reply pkgs/applications/graphics/pencil/default.nix]: Maybe, is it better ?

Comment (adisbladis) [reply pkgs/applications/graphics/pencil/default.nix]: `eudev` is the gentoo-maintained `udev` fork while `systemd` is the canonical `udev` upstream.

Comment (mrVanDalo) [reply pkgs/applications/graphics/pencil/default.nix]: @adisbladis Ah this explains why I thought it is the proper one :D (I used gentoo for a very long time in my life)

@Mic92 Yes it works, in a few minutes I'll push the changes.

**Pull request merged**